helix-core/src/main/java/org/apache/helix/monitoring/mbeans/CustomizedViewMonitor.java

```diff
@@ -0,0 +1,154 @@
+import org.apache.helix.monitoring.mbeans.dynamicMBeans.DynamicMetric;
+import org.apache.helix.monitoring.mbeans.dynamicMBeans.HistogramDynamicMetric;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CustomizedViewMonitor extends DynamicMBeanProvider {
+  private static final Logger LOG = LoggerFactory.getLogger(CustomizedViewMonitor.class);
+
+  private static final String MBEAN_DESCRIPTION = "Helix Customized View Aggregation Monitor";
+  private final String _clusterName;
+  private final String _sensorName;
+  private HistogramDynamicMetric _updateToAggregationLatencyGauge;
+  public static final String UPDATE_TO_AGGREGATION_LATENCY_GAUGE =
+      "UpdateToAggregationLatencyGauge";
+  private ClusterStatusMonitor _clusterStatusMonitor;
+
+  public CustomizedViewMonitor(String clusterName) {
+    _clusterName = clusterName;
+    _sensorName = String.format("%s.%s", MonitorDomainNames.AggregatedView.name(), _clusterName);
+    _updateToAggregationLatencyGauge =
+        new HistogramDynamicMetric(UPDATE_TO_AGGREGATION_LATENCY_GAUGE, new Histogram(
+            new SlidingTimeWindowArrayReservoir(getResetIntervalInMs(), TimeUnit.MILLISECONDS)));
+  }
+
+  @Override
+  public DynamicMBeanProvider register() throws JMException {
+    List<DynamicMetric<?, ?>> attributeList = new ArrayList<>();
+    attributeList.add(_updateToAggregationLatencyGauge);
+    doRegister(attributeList, MBEAN_DESCRIPTION, getMBeanName());
+    return this;
+  }
+
+  private ObjectName getMBeanName() throws MalformedObjectNameException {
+    return new ObjectName(String
+        .format("%s:%s=%s,%s=%s", MonitorDomainNames.AggregatedView.name(), "Type",
+            "CustomizedView", "Cluster", _clusterName));
+  }
+
+  @Override
+  public String getSensorName() {
+    return _sensorName;
+  }
+
+  void recordUpdateToAggregationLatency(long latency) {
+    if (_updateToAggregationLatencyGauge != null) {
+      _updateToAggregationLatencyGauge.updateValue(latency);
+    }
+  }
+
+  /**
+   * Find updated customized states and report the aggregation latency of each customized state
+   * Latency reporting excludes: updating a customized state that is the same as previous customized state,
+   * deleting a customized state
+   * @param updatedCustomizedViews Customized views that have been updated, obtained from CustomizedStateOutput
+   * @param curCustomizedViews Current customized view values from the CustomizedViewCache
+   * @param updatedStartTimestamps All customized state START_TIME property values from CustomizedStateOutput.
+   * START_TIME field is automatically updated when a customized state for a partition is updated by CustomizedStateProvider.
+   * @param updateSuccess If the customized view update to ZK is successful or not
+   * @param endTime The timestamp when the new customized view is updated to ZK
+   */
+  public void reportLatency(List<CustomizedView> updatedCustomizedViews,
```

Comment:
I discussed with @zhangmeng916 , we think global latency is good enough for our purpose. So we won't use per-resource latency since it can cause complexities and won't bring additional benefit for us in the foreseeable future.